src/Chapter3.hs

```diff
@@ -68,417 +67,482 @@ Haskell has several different ways to create entirely new data types. Let's talk
-  the same time. In contrast, for the sum types, we use OR to tell only about a
-  single possibility. AND in logic corresponds to multiplication in math, and OR
-  corresponds to addition. You see that there is some math theory behind the
-  concept of data types in Haskell, and that's why they are called Algebraic Data
-  Types.
--}
-
-{- |
-=🛡= Product type
-
-Let's now see how the product data types look like in Haskell.
-
-Product type should have a type name, one type constructor (the function that
-lets you create the value of the type later) and the description of the fields
-it consists of in the view of types.
-
-When defining a custom type in Haskell, you use the __"data"__ keyword, write
-the __type name__ you come up with after it, and then after the "=" sign you
-specify the __constructor name__ followed by the __fields__.
-
-When in action, a custom data type could be used in the following use case.
-A definition of a data type for a knight with a name and number of victories
-can look like this:
-
-@
-      ┌─ type name
-      │
-      │       ┌─ constructor name (or constructor tag)
-      │       │
-data Knight = MkKnight String Int
- │                       │    │
- │                       └────┴─── types of fields
- │
- └ "data" keyword
-@
-
-♫ NOTE: The constructor can have the same name as the type itself. And in most
-  cases, they are indeed named identically. This is not a problem for Haskell,
-  because types live in the types namespace, and constructors live in the value
-  namespace. So there won't be any collisions and misunderstandings from the
-  compiler side. The names are unambiguous.
-
-You can use the constructor name, to create a value of the "Knight" type.
-
-@
-arthur :: Knight
-arthur = MkKnight "Arthur" 100
-@
-
-A constructor is just a function from fields to the type! You can verify this in GHCi:
-
-ghci> :t MkKnight
-MkNight :: String -> Int -> Knight
-
-As in a regular function, you need to provide a 'String' and an 'Int' to the
-'MkNight' constructor in order to get the full-fledged 'Knight'.
-
-Also, you can write a function that takes a Knight and returns its name.
-It is convenient to use pattern matching for that:
-
-@
-knightName :: Knight -> String
-knightName (MkKnight name _) = name
-@
-
-And you can extract the name in GHCi:
-
-ghci> knightName arthur
-"Arthur"
-
-It is comfy to have such getters for all types, so Haskell provides a syntax for
-defining __records__ — named parameters for the product data type fields.
-
-Records have similar syntax for defining in Haskell as (unnamed) ordinary
-product types, but fields are specified in the {} separated by a comma. Each
-field should have a name and a type in this form: 'fieldName :: FieldType'.
-
-The same definition of 'Knight' but with records should be written in the
-following way:
-
-@
-data Knight = MkKnight
-    { knightName      :: String
-    , knightVictories :: Int
+-- |
+-- =🛡= Type aliases
```

Comment:
Would it be possible for you to configure your autoformatter not to change the style of comments?
It's easier for us when the diff is only for solutions and not for the whole text :slightly_smiling_face:

Reply:
Did y'all use a specific formatter/do you know how to turn off autoformatting comments on VS code? If not, I'll just be more conscious of what I'm committing!

Reply:
We use the [stylish-haskell](https://github.com/jaspervdj/stylish-haskell) code formatter, it's quite nice :slightly_smiling_face: But at the same time, I didn't expect VSCode plugin to enable any formatters by default :thinking:

Reply:
I updated the PR to just include the actual code changes! While we're here, this isn't technically related, but there's a couple things that look like they should be markdown formatted (like where it says "You are on __Chapter X__"); is that able to be formatted as well, and how would I set that up in VS Code?

Reply:
Unfortunately, comments in Haskell are not in Markdown. They are using a special syntax. The docs are called Haddock and you can generate full docs out of the comments. I am not sure about the plugin, as I am not a VSCode user, but I am sure, that it would be possible to generate in there.